packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-model.ts

```diff
@@ -0,0 +1,239 @@
+   * The name of the new model.
+   */
+  readonly modelName: string;
+  /**
+   * The definition of the primary docker image containing inference code, associated artifacts,
+   * and custom environment map that the inference code uses when the model is deployed for predictions.
+   */
+  readonly primaryContainer: IContainerDefinition;
+  /**
+   * Specifies the containers in the inference pipeline.
+   *
+   * @default - None
+   */
+  readonly containers?: IContainerDefinition[];
+
+  /**
+   * Isolates the model container. No inbound or outbound network calls can be made to or from the model container.
+   *
+   * @default false
+   */
+  readonly enableNetworkIsolation?: boolean;
+
+  /**
+   * A VpcConfig object that specifies the VPC that you want your model to connect to.
+   *
+   * @default - None
+   */
+  readonly vpcConfig?: VpcConfig;
+
+  /**
+   * Tags to be applied to the model.
+   *
+   * @default - No tags
+   */
+  readonly tags?: sfn.TaskInput;
+}
+
+/**
+ * A Step Functions Task to create a SageMaker model
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-sagemaker.html
+ * @experimental
+ */
+export class SageMakerCreateModel extends sfn.TaskStateBase implements iam.IGrantable, ec2.IConnectable {
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+  ];
+  /**
+   * Allows specify security group connections for instances of this fleet.
+   */
+  public readonly connections: ec2.Connections = new ec2.Connections();
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+  private readonly vpc?: ec2.IVpc;
+  private securityGroup?: ec2.ISecurityGroup;
+  private readonly securityGroups: ec2.ISecurityGroup[] = [];
+  private readonly subnets?: string[];
+  private readonly integrationPattern: sfn.IntegrationPattern;
+  private _role?: iam.IRole;
+  private _grantPrincipal?: iam.IPrincipal;
+
+  constructor(scope: cdk.Construct, id: string, private readonly props: SageMakerCreateModelProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern || sfn.IntegrationPattern.REQUEST_RESPONSE;
+    validatePatternSupported(this.integrationPattern, SageMakerCreateModel.SUPPORTED_INTEGRATION_PATTERNS);
+
+    // add the security groups to the connections object
+    if (props.vpcConfig) {
+      this.vpc = props.vpcConfig.vpc;
+      this.subnets = props.vpcConfig.subnets ? this.vpc.selectSubnets(props.vpcConfig.subnets).subnetIds : this.vpc.selectSubnets().subnetIds;
+    }
+
+    this.taskPolicies = this.makePolicyStatements();
+  }
+
+  /**
+   * The execution role for the Sagemaker Create Model API.
+   *
+   * Only available after task has been added to a state machine.
+   */
+  public get role(): iam.IRole {
+    if (this._role === undefined) {
+      throw new Error('role not available yet--use the object in a Task first');
```

Comment:
when would users run into these `not available yet` errors? what would they need to do to reconcile?
should we simplify the language to direct them to add the task to a state machine?